working/macros/api/introspection.dart

```diff
@@ -50,8 +68,8 @@ abstract class TypeDeclaration implements Declaration {
   /// The type parameters defined for this type declaration.
   Iterable<TypeParameterDeclaration> get typeParameters;
 
-  /// Create a type annotation representing this type with [typeArguments].
-  TypeAnnotation instantiate({List<TypeAnnotation> typeArguments});
+  /// Create a static type representing this type with [typeArguments].
+  StaticType instantiate({List<StaticType> typeArguments});
```

Comment:
We probably also want a `nullabilitySuffix`, to instantiate `List<int>` vs. `List<int>?`.

Reply:
Ah right good point, we have this on the `TypeAnnotation` but also need it on `StaticType`.

This should probably also be `async` I suppose :scream: